main.go

```diff
@@ -166,6 +166,46 @@ func handler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Validate whether the access token contains one of the request's accepted roles, if available in request header
+	acceptedRolesParam := r.Header.Get("X-Forward-Auth-Accepted-Roles")
+	if acceptedRolesParam != "" {
+		logger.Debug("validating accepted roles for request: " + acceptedRolesParam)
+		acceptedRoles := strings.Split(acceptedRolesParam, ",")
+		tokenClaims := make(map[string]interface{})
+		claimsBytes, err := payloadBytesFromJwt(storedToken.AccessToken)
+		if err != nil {
+			logger.Error("unable to parse token claims")
+			http.Error(w, "Internal server error", 500)
+		}
+		if err := json.Unmarshal(claimsBytes, &tokenClaims); err != nil {
+			logger.Error("unable to parse token claims")
+			http.Error(w, "Internal server error", 500)
+		}
+		
+		accessTokenRolesString, hasRoles := tokenClaims[fw.AccessTokenRolesField].(string)
+		if !hasRoles {
+			logger.Info("access token has no roles defined. not authorized.")
+			http.Error(w, "Not authorized", 401)
+		}	
+
+		accessTokenRoles := strings.Split(accessTokenRolesString, fw.AccessTokenRolesDelimiter)
+		hasRequiredRole := false
+		loopTokenRoles:
+		for _, tokenRole := range accessTokenRoles {
+			for _, acceptedRole := range acceptedRoles {
+				if tokenRole == acceptedRole {
+					hasRequiredRole = true
+					break loopTokenRoles
```

Comment:
i would make this a func and just return true when you find what you need, then you don't need to name and break the loop like this

eg.,

```
func hasRequiredRole(accessTokenRoles []string, acceptedRoles []string) bool {
    for _, tokenRole := range accessTokenRoles {
			for _, acceptedRole := range acceptedRoles {
				if tokenRole == acceptedRole {
					return true
				}
			}
		}
    return false
}
```